dill/_objects.py

```diff
@@ -111,6 +109,7 @@ class _Struct(ctypes.Structure):
         pass
     _Struct._fields_ = [("_field", ctypes.c_int),("next", ctypes.POINTER(_Struct))]
 _filedescrip, _tempfile = tempfile.mkstemp('r') # deleted in cleanup
+os.close(_filedescrip)
```

Comment:
There's probably a reason I didn't do that.  I'm assuming that it changes the file descriptor from an open one to a closed one… and those pickle differently.  I'll have to take a closer look, but please punt this to a new PR, and I can merge the rest of this.